<?xml version="1.0"?>
<launch>
  <arg name="uav_odom_topic_name" default="/ground_truth/state" />
  <arg name="yaw_mode" default="false" />

  ##### Perception Node #####
  <!-- 2D laser base -->
  <include file="$(find jsk_uav_forest_perception)/launch/laser_process.launch">
  </include>

  <!-- tree detector -->
  <node pkg="jsk_uav_forest_perception" type="tree_tracking_node" output="screen" name="tree_tracking_node">
    <param name="laser_scan_topic_name" value="/scan_clustered" />
    <param name="uav_odom_topic_name" value="/ground_truth/state" />
    <param name="verbose" value="false" />
  </node>

  ##### Motion Node #####
  <node pkg="jsk_uav_forest_motion" type="spline_path_node" name="aggressive_motion" output="screen" >
    <param name="yaw_mode" value="$(arg yaw_mode)"/>
    <param name="uav_odom_sub_topic_name" value="$(arg uav_odom_topic_name)"/>
    <param name="spline_degree" value="2"/>
    <param name="spline_segment_time" value="1.0"/>
    <param name="uav_cmd_traj_track_p_gain" value="0.05"/>
    <param name="uav_cmd_traj_track_i_gain" value="0.01"/>
    <param name="uav_cmd_traj_yaw_p_gain" value="1.0"/>
  </node>

  <node pkg="jsk_uav_forest_motion" type="active_plannar_node" name="active_plannar_node" output="screen">
    <param name="control_period" value="0.99"/>
    <param name="plan_period" value="1.0"/>
    <param name="goal_x" value="10.0"/>
    <param name="goal_y" value="0.0"/>
    <param name="goal_z" value="5.0"/>
    <param name="velocity_upper_bound" value="4.0"/>
    <param name="acceleration_upper_bound" value="2.0"/>
    <param name="transfer_cost_weight" value="0.0"/>
    <param name="velocity_differ_cost_weight" value="0.0"/>
  </node>
</launch>
